### PR TITLE
Closes #8415: Adds Neutrik power connectors

### LINKED ISSUE #8415: Add Neutrik powerCON power ports
### NetBox version

v3.1.6

### Feature type

New functionality

### Proposed functionality

Add Neutrik powerCON type connectors to the power port types:

- powerCON TRUE 1 TOP
- powerCON TRUE1
- powerCON 20A
- powerCON 32A
- 
https://www.neutrik.us/en-us/products/power

### Use case

Some of the Luminex GigaCore switches use these types of power connectors.

### Database changes

_No response_

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by JoeIzzard**
### Fixes: #8415
Adds support for Neutrik Power connectors:
- powerCon
- powerCon 32A
- powerCon TRUE1
- powerCon TRUE1 TOP

### CODE REVIEW

File: netbox/dcim/choices.py [hunk 1]
```diff
@@ -345,6 +345,10 @@ class PowerPortTypeChoices(ChoiceSet):
     TYPE_DC = 'dc-terminal'
     # Proprietary
     TYPE_SAF_D_GRID = 'saf-d-grid'
+    TYPE_NEUTRIK_POWERCON = 'neutrik-powercon'
+    TYPE_NEUTRIK_POWERCON_32A = 'neutrik-powercon-32'
+    TYPE_NEUTRIK_TRUE1 = 'neutrik-true1'
+    TYPE_NEUTRIK_TRUE1_TOP = 'neutrik-true1-top'
```
Comment (jeremystretch) [inline netbox/dcim/choices.py:351]: The types listed [here](https://www.neutrik.com/en/products/power) are:

- powerCON 20A
- powerCON 32A
- powerCON TRUE1
- powerCON TRUE1 TOP

We should add exactly these to NetBox, with the same exact capitalization and noting the amperage where indicated. Also be sure that the constant value includes "POWERCON" in its name as that is part of the identifier (e.g. `TYPE_NEUTRIK_POWERCON_TRUE1`).

```suggestion
    TYPE_NEUTRIK_POWERCON_20A = 'neutrik-powercon-20a'
    TYPE_NEUTRIK_POWERCON_32A = 'neutrik-powercon-32a'
    TYPE_NEUTRIK_POWERCON_TRUE1 = 'neutrik-powercon-true1'
    TYPE_NEUTRIK_POWERCON_TRUE1_TOP = 'neutrik-powercon-true1-top'
```

File: netbox/dcim/choices.py [hunk 2]
```diff
@@ -456,6 +460,10 @@ class PowerPortTypeChoices(ChoiceSet):
         )),
         ('Proprietary', (
             (TYPE_SAF_D_GRID, 'Saf-D-Grid'),
+            (TYPE_NEUTRIK_POWERCON, 'Neutrik powerCon'),
+            (TYPE_NEUTRIK_POWERCON_32A, 'Neutrik powerCon (32A)'),
+            (TYPE_NEUTRIK_TRUE1, 'Neutrik powerCon TRUE1'),
+            (TYPE_NEUTRIK_TRUE1_TOP, 'Neutrik powerCon TRUE1 TOP'),
```
Comment (jeremystretch) [inline netbox/dcim/choices.py:466]:
```suggestion
            (TYPE_NEUTRIK_POWERCON, 'Neutrik powerCON (20A)'),
            (TYPE_NEUTRIK_POWERCON_32A, 'Neutrik powerCON (32A)'),
            (TYPE_NEUTRIK_TRUE1, 'Neutrik powerCON TRUE1'),
            (TYPE_NEUTRIK_TRUE1_TOP, 'Neutrik powerCON TRUE1 TOP'),
```

Comment (jeremystretch) [reply netbox/dcim/choices.py:466]: Please make these edits for the outlet types below as well.

**Review by jeremystretch — CHANGES_REQUESTED** [review]

**Comment by JoeIzzard** [comment]
I have made those changes, sorry I had used a mix of the official name and the name that they are most commonly referred to as.

**Pull request merged**